src/session/types/AppSession.ts

```diff
@@ -8,4 +11,28 @@ export type AppSession = {
   expiresAt: number
   refreshToken: string
   accessToken: string
+  region: Region
 }
+
+export const isAppSession = (data: unknown): data is AppSession =>
+  hasKey(data, 'spaceId') &&
+  typeof data.spaceId === 'number' &&
```

Comment:
This two appears a lot together, so it could become `hasKey(data, 'spaceId', 'number')` with the third parameter being optional. Just a little suggestion, but feel free to skip.

Reply:
I have been thinking the same actually.

Though it does not cover the case when the property is an object or array

```
hasKey(data, 'spaceId') &&
  isArray(data.spaceId)
```  

But maybe 

```
hasKey(data, 'spaceId', 'array')
```

or

```
hasStringProp()
hasNumberProp()
hasBooleanProp()
hasArrayProp()
```

Actually, now with Typescript v5.0, Typescript correctly infers the type with the `in` operator, so the whole motivation for this utility function disappears:

```
typeof data === 'object' && 
'spaceId' in data &&
typeof data. spaceId === 'string' &&
```